admin/post-types/writepanels/writepanel-certificate_data.php

```diff
@@ -68,248 +68,76 @@ function certificate_template_data_meta_box( $post ) {
-							'id'    => '_certificate_course_pos',
-							'class' => 'field_pos',
-							'value' => implode( ',', $woothemes_sensei_certificate_templates->get_field_position( 'certificate_course' ) ),
-						)
-					);
-					certificate_templates_wp_font_select(
-						array(
-							'id'      => '_certificate_course',
-							'label'   => __( 'Font', 'sensei-certificates' ),
-							'options' => $available_fonts,
-						)
-					);
-					certificates_wp_text_input(
-						array(
-							'id'    => '_certificate_course_font_color',
-							'label' => __( 'Font color', 'sensei-certificates' ),
-							'value' => isset( $woothemes_sensei_certificate_templates->certificate_template_fields['certificate_course']['font']['color'] ) ? $woothemes_sensei_certificate_templates->certificate_template_fields['certificate_course']['font']['color'] : '',
-							'class' => 'colorpick',
-						)
-					);
-					certificates_wp_text_input(
-						array(
-							'class'       => 'medium',
-							'id'          => '_certificate_course_text',
-							'label'       => __( 'Course Text', 'sensei-certificates' ),
-							'description' => __( 'Text to display in the course area.', 'sensei-certificates' ),
-							'placeholder' => __( '{{course_title}}', 'sensei-certificates' ),
-							'value'       => isset( $woothemes_sensei_certificate_templates->certificate_template_fields['certificate_course']['text'] ) ? $woothemes_sensei_certificate_templates->certificate_template_fields['certificate_course']['text'] : '',
-						)
-					);
-				echo '</div>';
-
-				// Certificate complete position.
 				echo '<div class="options_group">';
 					certificate_templates_wp_position_picker(
 						array(
-							'id'          => 'certificate_completion_pos',
-							'label'       => __( 'Completion Date Position', 'sensei-certificates' ),
-							'value'       => implode( ',', $woothemes_sensei_certificate_templates->get_field_position( 'certificate_completion' ) ),
-							'description' => __( 'Optional position of the Course Completion date', 'sensei-certificates' ),
+							'id'          => "certificate_{$field_key}_pos",
+							'label'       => $field_info['position_label'],
+							'value'       => implode( ',', $woothemes_sensei_certificate_templates->get_field_position( "certificate_$field_key" ) ),
+							'description' => $field_info['position_description'],
 						)
 					);
 					certificates_wp_hidden_input(
 						array(
-							'id'    => '_certificate_completion_pos',
+							'id'    => "_certificate_{$field_key}_pos",
 							'class' => 'field_pos',
-							'value' => implode( ',', $woothemes_sensei_certificate_templates->get_field_position( 'certificate_completion' ) ),
+							'value' => implode( ',', $woothemes_sensei_certificate_templates->get_field_position( "certificate_$field_key" ) ),
 						)
 					);
 					certificate_templates_wp_font_select(
 						array(
-							'id'      => '_certificate_completion',
+							'id'      => "_certificate_$field_key",
```

Comment:
I think it would be better to wrap the variable with curly braces here as well.
First, to be consistent. Second, my first idea was that we use a `$field` variable and I didn't find it (so for clarity).

```suggestion
							'id'      => "_certificate_{$field_key}",
```